azurerm/resource_arm_application_gateway.go

```diff
@@ -1263,6 +1279,7 @@ func expandApplicationGatewayBackendHTTPSettings(d *schema.ResourceData, gateway
 			ApplicationGatewayBackendHTTPSettingsPropertiesFormat: &network.ApplicationGatewayBackendHTTPSettingsPropertiesFormat{
 				CookieBasedAffinity:            network.ApplicationGatewayCookieBasedAffinity(cookieBasedAffinity),
 				Path:                           utils.String(path),
+				HostName:                       utils.String(hostName),
```

Comment:
Do we always want to set this? how does the API behave when pick hostname is true?

Reply:
No, I don't think so. My mistake. I'm sending a fix right now.